src/main/java/backend/Logic.java

```diff
@@ -157,37 +160,84 @@ public void refresh() {
         for (Map.Entry<Integer, IssueMetadata> entry : metadata.entrySet()) {
             IssueMetadata currentMetadata = entry.getValue();
             Date lastNonSelfUpdate = new Date(0);
+            Date lastSelfUpdate = new Date(0);
+            boolean isUpdatedByOthers = false;
+            boolean isUpdatedBySelf = false;
             for (TurboIssueEvent event : currentMetadata.getEvents()) {
-                if (!event.getActor().getLogin().equalsIgnoreCase(currentUser)
-                        && event.getDate().after(lastNonSelfUpdate)) {
+                if (isNewEventByOther(currentUser, event, lastNonSelfUpdate)) {
                     lastNonSelfUpdate = event.getDate();
+                    isUpdatedByOthers = true;
```

Comment:
All this logic is about the (re-)construction of `IssueMetadata` given the current user (which isn't available deeper in the back end). Could it be moved into a new `IssueMetadata` constructor? `new IssueMetadata(existingMetadata, currentUser)`